Add wpt tests for send

dlrobertson opened the pull request:
Add wpt tests for send that do not set the responseType and compare the result against a string.

Fixes #9357 

r? @KiChjang

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/9971)

<!-- Reviewable:end -->

danlrobertson commented:
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/servo/servo/9971#-KCbk6FvVnH6KTJN049E:-KCbk6FwDr3b686dRixa:1964149046) (no related file):*
I was confused as to what tests should be added, so I based it off of the comment by @jdm. If you would like tests for `URLSearchParams` etc, just let me know

---

*[tests/wpt/mozilla/meta/MANIFEST.json, line 6110 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9971#-KCbj4Yx5gva8HScMFuL:-KCbj4YyXtjXpsjkc9UA:1359212003) ([raw file](https://github.com/servo/servo/blob/c6093f3a31a03e5add03ac74d05c44e301c3b068/tests/wpt/mozilla/meta/MANIFEST.json#L6110)):*
These were the only two tests among `tests/wpt/web-platform-tests/XMLHttpRequest/send-data-*` that either set the responseType (`blob`) or would fail without comparing the data against a string (`arraybuffer`).

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9971)*
<!-- Sent from Reviewable.io -->

KiChjang commented:
We're not supposed to add in WPT tests, per the discussion in the issue. Instead, we need unit tests, and they're written in Rust.

jdm commented:
No, this PR is doing what was requested in the original issue - we're adding servo-specific WPT tests that don't get upstreamed that allow us to test the things we care about without relying on features we don't implement yet.

danlrobertson commented:
So what should I do about `resources/content.py`? the lint requests a license, but the file is copied from `web-platform-tests`.

If licensing is a problem, I can always just make a new `content.py` and add our license header.

KiChjang commented on tests/wpt/mozilla/meta/MANIFEST.json:
```diff
@@ -6096,6 +6096,18 @@
+            "path": "mozilla/send-without-content-type-blob.htm",
+            "url": "/_mozilla/mozilla/send-without-content-type-blob.htm"
+          }
+        ],
```
Is this manually typed? Tidy doesn't like the indentation of the brackets.

danlrobertson replied:
Good point! Yes it was, updated using `mach create-wpt`

KiChjang commented:
Just one small nit - why are the test files named *-without-content-type-*?

danlrobertson commented:
Ah! Sorry, should have been send-without-response-type-*. Let me know if you prefer another name.

---

Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9971#-:-KD-EmBSldzedQcDykev:-1011028422)*
<!-- Sent from Reviewable.io -->

KiChjang commented:
This still is a misnomer - the response being parsed is a text response. I suggest removing the -without-response-type altogether since this isn't relevant to the test that you have wrote.

dlrobertson committed:
Add wpt tests for send

Add wpt tests for send that do not set the responseType and
compare the result against a string.

danlrobertson commented:
True, changed it to just `send-*`.

KiChjang commented on tests/wpt/mozilla/tests/mozilla/send-arraybuffer.htm:
```diff
@@ -0,0 +1,49 @@
+            };
+
+            xhr.open("POST", "./resources/content.py", true);
+            xhr.responseType = "text";
```
We shouldn't be explicitly setting the responseType here - we should instead assert that the response we read from xhr is a text string.

KiChjang commented:
I'm unsure about whether we should include tests for `URLSearchParams`. If we do include it, I think it should be a WPT test instead of a mozilla test.

danlrobertson commented:
> I'm unsure about whether we should include tests for URLSearchParams. If we do include it, I think it should be a WPT test instead of a mozilla test.

Makes sense!

I removed the `responseType = "text"`.

KiChjang commented:
I'm going to go ahead and merge this. Filed w3c/web-platform-tests#2709 for the missing test coverage. Thanks for working on this!

@bors-servo r+

bors-servo commented:
:pushpin: Commit 8ca7ff1 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang 8ca7ff11ca693ee541bdbcb43d097312cba09935 -->

bors-servo commented:
:hourglass: Testing commit 8ca7ff1 with merge be330b5...

bors-servo commented:
:broken_heart: Test failed - [android](http://build.servo.org/builders/android/builds/2047)

Ms2ger commented:
@bors-servo retry
* #10071

bors-servo commented:
:hourglass: Testing commit 8ca7ff1 with merge 14c6497...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/2049), [gonk](http://build.servo.org/builders/gonk/builds/2105), [linux-dev](http://build.servo.org/builders/linux-dev/builds/2083), [linux-rel](http://build.servo.org/builders/linux-rel/builds/2364), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/776), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/2132), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/2190), [status-appveyor](https://ci.appveyor.com/project/servo/servo/build/1.0.637)
